### PR TITLE
[IMP] mail: Add tests + some fixes in mobile view

### PR DESCRIPTION
**Pull request by maki-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/messaging_menu/messaging_menu.xml [hunk 1]
```diff
@@ -101,7 +103,7 @@
             'text-primary fw-bolder o-active': store.discuss.activeTab === tab.id,
             'border-end': !tab_last,
         }" t-on-click="() => this.onClickNavTab(tab.id)">
-            <i t-attf-class="p-2 fs-4 {{ tab.icon }}"/>
+            <i t-attf-aria-label="{{ tab.id }}" t-attf-class="p-2 fs-4 {{ tab.icon }}"/>
```
Comment (alexkuhn) [inline addons/mail/static/src/new/messaging_menu/messaging_menu.xml]: Do not introduce this aria label. It makes no sense for a casual end-user, and the tab.label is visually present so you can rely on this for the selectors in test

File: addons/mail/static/src/new/messaging_menu/messaging_menu.xml [hunk 2]
```diff
@@ -26,17 +26,19 @@
                 <button class="btn btn-link" t-att-class="store.discuss.activeTab === 'channel' ? 'fw-bolder' : 'text-500'" type="button" role="tab" t-on-click="() => store.discuss.activeTab = 'channel'">Channels</button>
             </t>
             <div class="flex-grow-1"></div>
-            <t t-if="!store.discuss.isActive or store.isSmall and env.inDiscussApp and ['chat', 'channel'].includes(store.discuss.activeTab)">
-                <div t-if="state.addingChat or state.addingChannel" class="p-4 border-bottom bg-light">
-                    <t t-if="state.addingChat" t-set="category" t-value="store.discuss.chats"/>
-                    <t t-if="state.addingChannel" t-set="category" t-value="store.discuss.channels"/>
-                    <ChannelSelector category="category" autofocus="true"/>
+            <t t-if="!store.discuss.isActive or store.isSmall and store.discuss.activeTab !== 'mailbox'">
+                <div t-if="env.inDiscussApp and ['chat', 'channel'].includes(store.discuss.activeTab)">
```
Comment (alexkuhn) [inline addons/mail/static/src/new/messaging_menu/messaging_menu.xml]: `<div` can become `<t`. Ok to split in 2 logic expression as to not mix `and` and `or` together in a single line.

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 3]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+        );
+        await click(".o-mail-channel-selector-suggestion");
+        await triggerEvent(target, ".o-mail-channel-selector-input", "keydown", {
+            key: "Enter",
+        });
+        assert.containsOnce(target, ".o-mail-chat-window-header-name[title='Gandalf']");
+    }
+);
+QUnit.test(
+    "Choose an available channel or create a new one via channel selector in discuss in small device should open a chat window",
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: Name of tests are far too long. Could be more brief

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 4]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+        await openDiscuss();
+        await click("i[aria-label='channel']");
+        assert.containsOnce(target, "button:contains(New Channel)");
+        assert.containsNone(target, "input[placeholder='Add or join a channel']");
+        await click("button:contains(New Channel)");
+        assert.containsOnce(target, "input[placeholder='Add or join a channel']");
+        assert.containsNone(target, "button:contains(New Channel)");
+        await afterNextRender(() => triggerEvent(target, ".o-mail-messaging-menu", "click"));
+        assert.containsOnce(target, "button:contains(New Channel)");
+        assert.containsNone(target, "input[placeholder='Add or join a channel']");
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: Don't make it look like with same shape of `once`/`none`. Instead keep the ordering of asserted selectors the same.

It makes it much easier to see change from `Once` to `None` on the same selector at a glance.

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 5]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+        patchUiSize({ height: 360, width: 640 });
+        const { openDiscuss } = await start();
+        await openDiscuss();
+        await click("i[aria-label='chat']");
+        assert.containsOnce(target, "button:contains(Start a conversation)");
+        assert.containsNone(target, "input[placeholder='Start a conversation']");
+        await click("button:contains(Start a conversation)");
+        assert.containsOnce(target, "input[placeholder='Start a conversation']");
+        assert.containsNone(target, "button:contains(Start a conversation)");
+        await afterNextRender(() => triggerEvent(target, ".o-mail-messaging-menu", "click"));
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: Could just `await click(".o-mail-messaging-menu")`

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 6]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+);
+
+QUnit.test(
+    "'New Message' button should open a chat window in small and non small devices",
+    async function (assert) {
+        const { env, openView } = await start();
+        await openView({
+            res_model: "res.partner",
+            views: [[false, "form"]],
+        });
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: There's no reason to open form view in this test

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 7]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+        await click(".o_menu_systray i[aria-label='Messages']");
+        await click(".o_menu_systray button:contains(New Message)");
+        assert.containsOnce(target, ".o-mail-chat-window");
+        await click(".o-mail-command[title='Close chat window']");
+        assert.containsNone(target, ".o-mail-chat-window");
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: You should not test the chat window is closed from "Close chat window". This is already covered by another test.

You should avoid making test that assert everything, otherwise we have far too many tests that are unmaintainable. Ideally a test should assert 1 and only 1 thing precisely. When several assertions, they should all address a same specific thing being tested.

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 8]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+    "'New Message' button should open a chat window in small and non small devices",
+    async function (assert) {
+        const { env, openView } = await start();
+        await openView({
+            res_model: "res.partner",
+            views: [[false, "form"]],
+        });
+        await click(".o_menu_systray i[aria-label='Messages']");
+        await click(".o_menu_systray button:contains(New Message)");
+        assert.containsOnce(target, ".o-mail-chat-window");
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: Non-mobile "new message" is already tested. Keep test specific to testing 1 and only 1 thing.
We might have duplicates, and it's fine to clean them at a later date. But do not add test for the sake of adding test, otherwise we cannot clean by removing duplicates :')

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 9]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+        );
+        await click(".o-mail-channel-selector-suggestion");
+        assert.containsOnce(target, ".o-mail-chat-window-header-name[title='Gryffindors']");
+        await click(".o-mail-chat-window-header div[title='Close conversation']");
+        await click("button:contains(New Channel)");
+        await afterNextRender(() =>
+            editInput(target, "input[placeholder='Add or join a channel']", "slytherins")
+        );
+        await click(".o-mail-channel-selector-suggestion");
+        assert.containsOnce(target, ".o-mail-chat-window-header-name[title='slytherins']");
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: Make 2 tests, rather than a long test with lots of user interactions that require coverage of other tests (e.g. closing a chat window)

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 10]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+        pyEnv["res.users"].create({ partner_id: partnerId });
+        const { openDiscuss } = await start();
+        await openDiscuss();
+        await click("i[aria-label='chat']");
+        await click("button:contains(Start a conversation)");
+        await afterNextRender(() =>
+            editInput(target, "input[placeholder='Start a conversation']", "Gandalf")
+        );
+        await click(".o-mail-channel-selector-suggestion");
+        await triggerEvent(target, ".o-mail-channel-selector-input", "keydown", {
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: I prefer using `triggerHotkey`, it's shorter and easier to read

File: addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js [hunk 11]
```diff
@@ -437,6 +437,107 @@ QUnit.test("open chat window from preview", async function (assert) {
...
+        const pyEnv = await startServer();
+        const partnerId = pyEnv["res.partner"].create({ name: "Gandalf" });
+        pyEnv["res.users"].create({ partner_id: partnerId });
+        const { openDiscuss } = await start();
+        await openDiscuss();
+        await click("i[aria-label='chat']");
+        await click("button:contains(Start a conversation)");
+        await afterNextRender(() =>
+            editInput(target, "input[placeholder='Start a conversation']", "Gandalf")
+        );
```
Comment (alexkuhn) [inline addons/mail/static/tests/new/messaging_menu/messaging_menu_tests.js]: `insertText` is nicer here than `editInput`
